modules/Bio/EnsEMBL/BaseAlignFeature.pm

```diff
@@ -1009,7 +1012,266 @@ sub _query_unit {
   throw( "Abstract method call!" );
 }
 
+=head2 _mdtag_alignment_length
+
+  Arg [1]    : None
+  Description: return the alignment length (including indels) based on the mdtag (mdz) string
+  Returntype : int
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+
+sub _mdtag_alignment_length {
+  my $self = shift;
+
+  if (! defined $self->{'_alignment_length'} && defined $self->cigar_string) {
+    my $mdz_string =  $self->cigar_string;
+    my $chunks = $self->_get_mdz_chunks($mdz_string);
+    my $alignment_length = 0;
+    $alignment_length = $self->_get_mdz_alignment_length($chunks);
+    $self->{'_alignment_length'} = $alignment_length;
+  }
+  return $self->{'_alignment_length'};
+}
+
+=head2 _get_mdz_chunks
+
+  Arg [1]    : mdtag string - MD Z String for mismatching positions. Regex : [0-9]+(([A-Z]|\^[A-Z]+)[0-9]+)* (Refer:  SAM/BAM specification)
+  Description: parses the mdtag string and group it according the type
+               eg: MD:Z:35^VIVALE31^GRPLIQPRRKKAYQLEHTFQGLLGKRSLFTE10 returns ['35', '^', 'VIVALE', '31', '^', 'GRPLIQPRRKKAYQLEHTFQGLLGKRSLFTE', '10']
+  Returntype : array of strings
+  Exceptions : none
+  Caller     : internal
+  Status     : Stable
+
+=cut
+sub _get_mdz_chunks {
+  my $self = shift;
+
+  my $mdz_string = shift;
+  my $alignment_length = 0;
+  my @chunks;
+    if ( $mdz_string =~ /^MD:Z:(.*)/ ){
+      my $mdtag = $1;
+
+      #if start to end is a number then all match return as it is
+      if($mdtag =~/^\d+$/){
+        return [$mdtag];
+      }else{
+         my @char_arr = split('',$mdtag);
+         my $cur_char = undef;
+
+         while(defined($cur_char = shift @char_arr)){
```

Comment:
Is the defined needed? Won't shift on an empty array get undef and eval as false?